### PR TITLE
ANN: don't annotate unknown types in format macros

### PR DESCRIPTION
**Pull request by Kobzol**
Fixes: https://github.com/intellij-rust/intellij-rust/issues/5378

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsFormatMacroAnnotator.kt [hunk 1]
```diff
@@ -411,7 +412,7 @@ private fun checkParameterTraitMatch(argument: RsFormatMacroArg, parameter: Form
     val requiredTrait = parameter.type?.resolveTrait(argument.knownItems) ?: return null
 
     val expr = argument.expr
-    if (!expr.implLookup.canSelectWithDeref(TraitRef(expr.type, requiredTrait.withSubst()))) {
+    if (expr.type != TyUnknown && !expr.implLookup.canSelectWithDeref(TraitRef(expr.type, requiredTrait.withSubst()))) {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsFormatMacroAnnotator.kt:415]:
```suggestion
    if (!expr.type.containsTyOfClass(TyUnknown::class.java) && !expr.implLookup.canSelectWithDeref(TraitRef(expr.type, requiredTrait.withSubst()))) {
```

Comment (Kobzol) [reply src/main/kotlin/org/rust/ide/annotator/RsFormatMacroAnnotator.kt:415]: Thanks.

**Comment by Kobzol** [comment]
Should I also ignore `Never` (!) types? I'm not sure why anyone would want to print a ! type, but currently we show a false positive error for something like `println!("{}", std::process::exit(0));`.

**Comment by vlad20012** [comment]
> Should I also ignore Never (!) types? 

Probably yes. I wonder why it doesn't implement `Display `trait... Or how it works

**Comment by Kobzol** [comment]
Hmm, for some reason in tests `ImplLookup` claims that `!` implements `Display`, but in the IDE it doesn't. I explicitly ignore both `TyUnknown` and `TyNever` now.

**Pull request merged**